unified-scheduler-pool/src/lib.rs

```diff
@@ -1377,21 +1363,23 @@ impl<TH: TaskHandler> SpawnableScheduler<TH> for PooledScheduler<TH> {
     ) -> Self {
         inner
             .thread_manager
-            .start_session(&context, result_with_timings);
+            .start_session(context.clone(), result_with_timings);
```

Comment:
also, `.start_session()` and `.start_threads()` is really similar to each other.